### DISCUSSION TITLE
FR: Add more "magic" to magic areas

### DISCUSSION THREAD

**Discussion by xBourner** [jseidl/magic-areas#392] [Ideas / Feature Requests]
I'm coming from Dwains Dashboard (https://github.com/dwainscheeren/dwains-lovelace-dashboard) which is basically the same what magic areas do. The only negative things about Dwains is that he does not work on it anymore and that is not so much customizable (only if you change his js code - what i also did)

But still i always searched for a way to setup an auto configurating dash board or more some auto generated entites or groups which show certain stats of certain areas all on its own. This is where magic areas comes into place.
This integration almost does everything i want but does not cover every domain of entities.

For example im missing all kind of sensors (humidity, pressure, lux etc) or even default sensors and vacuum and switch entities.
It would be cool if magic areas would create a new group for every domain connected to the assigned area and meta areas. 
its a shame that HA isnt doing this on its own. Are you planning to implement more device classes/domains into magic areas?

**Comment by jseidl** [comment]
Hi, have you enabled the aggregates feature? Magic areas does create aggregates for every binary sensor or sensor you have. I wonder what I'm missing here.

**Reply by xBourner** [reply]
Yes of course. Sensor feature gives me temperature and batteries for example. But not humidity or pressure. Lumination (lux) is missing too.

all these entities come from the same sensors (aqara zigbee) and the aqara device is assigned to an area.

**Reply by mihsu81** [reply]
@xBourner in the configuration menu, under "Configure aggregate sensors" set to 1 the minimum number of entities.

**Reply by xBourner** [reply]
@mihsu81 oh man i feel so dumb. of course that will add my missing entities. then the only things i am missing for now are vacuum and switch domain. any plans in that @jseidl ?

**Reply by jseidl** [reply]
Vacuum could be interesting indeed, as for switches, it's hard to infer how to group them, as they might have different "categories". Can you cut new feature requests for those individually?

**Reply by xBourner** [reply]
Yeah i created two new FRs. One for vaccum and one for switch. 
Would be very nice to have kind of all domains in magic areas so every entity would be automatically assigned to areas. I don't understand how HA does not do that on default. Keep up the great work!

**Comment by tbrasser** [comment]
Years ago I was also a DD user, and now my dashboard/interface is a lovelace_gen/auto-entities/magic_areas combo.

We should be able to create strategies tailored to magic_areas: https://developers.home-assistant.io/docs/frontend/custom-ui/custom-strategy/

https://github.com/itsteddyyo/strategy-pack
